core/src/task.rs

```diff
@@ -536,6 +538,17 @@ macro_rules! vdaf_dispatch {
 #[derive(Clone)]
 pub struct AuthenticationToken(Vec<u8>);
 
+impl AuthenticationToken {
+    /// Constructs a bearer token string suitable for use as the value in an HTTP `Authorization`
+    /// header, or an error if this token is not valid UTF-8.
+    pub fn bearer_token(&self) -> Result<String, anyhow::Error> {
+        Ok(format!(
+            "Bearer {}",
+            str::from_utf8(self.as_ref())
+                .context("failed to covert authentication token to UTF-8")?
```

Comment:
I'm now taking a look at the places where we generate and store auth tokens across the aggregator API and the Janus API and TBH it's a bit of a mess of (unintentional?) double-base64-encoding. I'm going to take a beat to try to untangle this and separate what an auth token _is_ (opaque blob of bytes) vs. how it's _represented_ for transport via HTTP headers or k8s secrets or YAML or environment variables.

Reply:
Yes, I agree, and symmetrically, `extract_bearer_token` must base64 decode, since this particular layer of base64 encoding is a detail of how we transport the token in the context of an `authorization: Bearer` header.

Where this gets confusing is that the `impl Distribution<AuthenticationToken> for Standard` constructs `AuthenticationToken`s by generating 16 random bytes, then encodes that in Base64URL-no-padding, then the bytes of that string are the contents of the `AuthenticationToken`. So this worked because the bytes of the authentication token were themselves a valid UTF-8 and Base64 string. I'm not sure if it makes sense to generate `AuthenticationToken` that way (at a minimum it's a less efficient way to store 16 bytes of entropy) but that's orthogonal to the goals of this change, so I won't mess with it.

Reply:
Yeah -- I think the reason we made that choice was because the semantics of `DAP-Auth-Token` are "there is a correct token that must match for authz to pass, we pass it back and forth over an HTTP header" (which implies it must fit into an HTTP header, i.e. it is ASCII). So we chose to generate a random token by base64-encoding some random bytes, which is a handy way to create a token with a given amount of entropy that fits the ASCII requirement -- but in this case, the actual token **is** the base64-encoded string, not the underlying bytes that were randomly generated. And any process that could encode an ASCII string from an arbitrary byte buffer would work equally well, e.g. we could have hex-encoded, base32-encoded, etc.

Now that we're using `Authorization: Bearer`, the semantics are defined somewhat differently: we _must_ send data that is at least _shaped_ like base64-encoded data ([RFC](https://www.rfc-editor.org/rfc/rfc6750.html#section-2.1)). Since our authorization tokens are actually arbitrary byte strings, we need to base64-encode here.

Once we have migrated away from supporting `DAP-Auth-Token` at all, or refactored things such that we can generate auth tokens per-authorization-method, we can change our encoding. It's a bit unfortunate we defined `DAP-Auth-Token` so loosely--maybe we should change our encoding scheme to keep things from getting confusing in practice? Or maybe we should support receiving auth tokens on a per-authorization-scheme basis, to avoid locking ourselves into the pattern that every authorization scheme must work with the same set of arbitrary bytes (or, alternatively, that our generated tokens must fit the requirements of every authorization scheme)?